packages/azure/src/lib/ConsumptionManagement.ts

```diff
@@ -570,7 +582,11 @@ export default class ConsumptionManagementService {
       replicationFactor: this.getReplicationFactor(consumptionDetailRow),
     }
 
-    let estimate: FootprintEstimate
+    let estimate: FootprintEstimate = {
+      timestamp: new Date(),
+      kilowattHours: 0,
+      co2e: 0,
+    }
```

Comment:
Hi @4upz, so should I modify it so line 585 is
```
    const defaultEstimate: FootprintEstimate = {
      timestamp: new Date(),
      kilowattHours: 0,
      co2e: 0,
    }
    let estimate: FootprintEstimate = defaultEstimate;
```
    
and line 608 to be like below, so it's checking defaultEstimate is not the assigned value?

`    if(estimate && estimate !== defaultEstimate)`

Reply:
@abc2Since these are objects, we can't compare them using `===` as it will always return false since it does not do a deep assertion of the values of each object. We can probably check for the presence or truthy (non-zero) value of a co2e or kilowattHours property by doing:

```
if(estimate.co2e && estimate.kilowattHours)
```

Alternatively, you can use the Rambda library that is in our project to do a deep equals of two values. Assuming Ramdba is imported as `R`:
`if( R.equals(estimate, defaultEstimate)`